build-tools/packages/build-cli/src/commands/generate/node10Entrypoints.ts

```diff
@@ -0,0 +1,135 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fs from "node:fs/promises";
+
+import * as resolve from "resolve.exports";
+import type { PackageJson } from "type-fest";
+import { BaseCommand } from "../../library/index.js";
+// AB#8118 tracks removing the barrel files and importing directly from the submodules, including disabling this rule.
+// eslint-disable-next-line import/no-internal-modules
+import { readPackageJson, readTsConfig } from "../../library/package.js";
+// AB#8118 tracks removing the barrel files and importing directly from the submodules, including disabling this rule.
+// eslint-disable-next-line import/no-internal-modules
+import type { Node10CompatExportData } from "../../library/packageExports.js";
+import type { CommandLogger } from "../../logging.js";
+
+export default class GenerateNode10EntrypointsCommand extends BaseCommand<
```

Comment:
nit: not a blocker. Ideally either the path to the folder or package.json would be a flag, even if it has a default
value. Makes it easier to test. But it can always be added later.